lib/sparse/src/index/inverted_index/inverted_index_ram.rs

```diff
@@ -69,7 +69,13 @@ impl InvertedIndexRam {
     }
 
     /// Upsert a vector into the inverted index.
+    ///
+    /// Empty vectors are ignored.
     pub fn upsert(&mut self, id: PointOffsetType, vector: SparseVector) {
+        // do not upsert empty vectors
+        if vector.is_empty() {
```

Comment:
I propose to move this check to `SparseVectorIndex::update_vector`. I expect that only one entity is responsible for skipping empty vectors. It can be `SparseVectorIndex` or `InvertedIndex`. Because `build` function has already a skipping, I propose to move this skip

Reply:
You may also have two checks: here and in `SparseVectorIndex::update_vector` both

Reply:
I like the idea of having a single responsible entity for skipping :+1: 
https://github.com/qdrant/qdrant/pull/3057/commits/d8281a738793e16e22e4c3c8bf33ff67a5b78da3